codec-http/src/main/java/io/netty/handler/codec/http/websocketx/extensions/WebSocketExtensionFilter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2019 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http.websocketx.extensions;
+
+import io.netty.handler.codec.http.websocketx.WebSocketFrame;
+
+/**
+ * Filter that is responsible to skip the evaluation of a certain extension
+ * according to standard.
+ */
+public interface WebSocketExtensionFilter {
+
+    /**
+     * A {@link WebSocketExtensionFilter} that never skip the evaluation of an
+     * any given extensions {@link WebSocketExtension}.
+     */
+    WebSocketExtensionFilter NEVER_SKIP = new WebSocketExtensionFilter() {
```

Comment:
I'm curious if this is of any use for users. Could you show me some use case of `NEVER_SKIP`? The reason why I'm asking this question is because if it is going to be used only internally, then perhaps we should not expose this in our public API.

Reply:
For example: 
```
public class SelectivityCompressionFilterProvider implements WebSocketExtensionFilterProvider {

    private final int contentSizeThreshold;

    public SelectivityCompressionFilterProvider(int contentSizeThreshold) {
        this.contentSizeThreshold = contentSizeThreshold;
    }

    @Override
    public WebSocketExtensionFilter encoderFilter() {
        return new WebSocketExtensionFilter() {
            @Override
            public boolean mustSkip(WebSocketFrame frame) {
                if (contentSizeThreshold > 0) {
                    return (frame instanceof BinaryWebSocketFrame || 
                            frame instanceof TextWebSocketFrame) &&
                            frame.content().readableBytes() < contentSizeThreshold;
                }

                return false;
            }
        };
    }

    @Override
    public WebSocketExtensionFilter decoderFilter() {
        return WebSocketExtensionFilter.NEVER_SKIP;
    }
}

Reply:
Ah, that makes sense, Thanks for explanation.